packages/itwin/grouping-mapping-widget/src/components/Mappings/Import/SelectIModel.tsx

```diff
@@ -22,38 +23,26 @@ const defaultDisplayStrings = {
 };
 
 const fetchIModels = async (
-  setIModels: (iModels: IModel[]) => void,
-  setIsLoading: (isLoading: boolean) => void,
   getAccessToken: GetAccessTokenFn,
   iTwinId: string,
   iModelsClient: IModelsClient,
 ) => {
-  try {
-    setIModels([]);
-    setIsLoading(true);
-    const iModelIterator: EntityListIterator<IModel> = iModelsClient.iModels.getRepresentationList({
-      authorization: async () => {
-        const token = await getAccessToken();
-        const splitToken = token.split(" ");
-        return {
-          scheme: splitToken[0],
-          token: splitToken[1],
-        };
-      },
-      urlParams: {
-        iTwinId,
-      },
-    });
-    const iModels: IModel[] = [];
-    for await (const iModel of iModelIterator) {
-      iModels.push(iModel);
-    }
-    setIModels(iModels);
-  } catch (error: any) {
-    handleError(error.status);
-  } finally {
-    setIsLoading(false);
+  const accessToken = await getAccessToken();
+  const iModelIterator = iModelsClient.iModels.getRepresentationList({
```

Comment:
Does this really need the whole iModel representation? Maybe minimal would be enough?

Reply:
It needs it. The minimal only provides the display name and id. This table also shows the description of the iModel.